Homework4/DmytroFenevHW4.cpp

```diff
@@ -0,0 +1,215 @@
+#include <algorithm>
+#include <chrono>
+#include <cmath>
+#include <functional>
+#include <iostream>
+
+using Comp = bool (*)(int, int);
+bool Vniz(int a, int b) { return a > b; }
+bool Vverx(int a, int b) { return a < b; }
+
+template <typename T> void sort(T *begin, const T *const end, Comp) {
+  if (begin == nullptr || end == nullptr || begin == end) {
+    return;
+  }
+  for (int *it_level1 = begin; it_level1 != end; ++it_level1) {
+    for (int *it_level2 = it_level1 + 1; it_level2 != end; ++it_level2) {
+      if (Comp()) {
```

Comment:
вы должны в эту функцию, компартор передать значения которые будут сравниваться
```suggestion
      if (comp(*it_level1, *it_level2)) {
```